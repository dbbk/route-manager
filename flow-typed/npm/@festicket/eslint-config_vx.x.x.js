// flow-typed signature: 564086c0274f153b4c37ac5c954f219e
// flow-typed version: <<STUB>>/@festicket/eslint-config_v^1.1.5/flow_v0.63.1

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@festicket/eslint-config'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@festicket/eslint-config' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */


// Filename aliases
declare module '@festicket/eslint-config/index' {
  declare module.exports: $Exports<'@festicket/eslint-config'>;
}
declare module '@festicket/eslint-config/index.js' {
  declare module.exports: $Exports<'@festicket/eslint-config'>;
}
