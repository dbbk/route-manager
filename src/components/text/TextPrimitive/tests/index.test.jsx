import React from 'react';
import { mount } from 'enzyme';
import { TextPrimitive } from '@festicket/react-ui-components';

describe('<TextPrimitive />', () => {
  test('with initial props', () => {
    const tree = mount(<TextPrimitive>Test</TextPrimitive>);
    expect(tree).toMatchSnapshot();
  });
});

describe('<Text /> primitive', () => {
  test('with default props', () => {
    const wrapper = mount(<TextPrimitive>default</TextPrimitive>);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render correctly based on element prop', () => {
    const wrapper = mount(<TextPrimitive element="h3">default</TextPrimitive>);
    expect(wrapper).toMatchSnapshot();
  });

  test('should render correctly based on variant prop', () => {
    const wrapper = mount(
      <TextPrimitive element="h3" variant="h2">
        default
      </TextPrimitive>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render correctly based on size prop', () => {
    const wrapper = mount(
      <TextPrimitive element="p" size="tiny">
        default
      </TextPrimitive>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render correctly based on weight prop', () => {
    const wrapper = mount(
      <TextPrimitive element="p" weight="light">
        default
      </TextPrimitive>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render correctly based on color prop', () => {
    const wrapper = mount(
      <TextPrimitive element="p" color="grey">
        default
      </TextPrimitive>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render correctly based on underline prop', () => {
    const wrapper = mount(
      <TextPrimitive element="p" color="grey" underline>
        default
      </TextPrimitive>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
