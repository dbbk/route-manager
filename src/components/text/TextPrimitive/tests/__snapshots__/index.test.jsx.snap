// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> primitive extended 1`] = `
.c1 {
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  margin-bottom: 20px;
  text-decoration: none;
  font-size: 18px;
  line-height: 1.12;
}

.c0 {
  line-height: 2;
}

<StyledText>
  <TextPrimitive
    className="c0"
  >
    <Paragraph
      className="c0"
      color="black"
      element="p"
      size="regular"
      spacing="md"
      underline={false}
      weight="regular"
    >
      <Primitive
        className="c0 c1"
        color="black"
        element="p"
        size="regular"
        spacing="md"
        underline={false}
        weight="regular"
      >
        <p
          className="c0 c1"
        >
          extended text test
        </p>
      </Primitive>
    </Paragraph>
  </TextPrimitive>
</StyledText>
`;

exports[`<Text /> primitive should render correctly based on color prop 1`] = `
.c0 {
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  margin-bottom: 20px;
  text-decoration: none;
  font-size: 18px;
  line-height: 1.12;
}

<TextPrimitive
  color="grey"
  element="p"
>
  <Paragraph
    color="grey"
    element="p"
    size="regular"
    spacing="md"
    underline={false}
    weight="regular"
  >
    <Primitive
      className="c0"
      color="grey"
      element="p"
      size="regular"
      spacing="md"
      underline={false}
      weight="regular"
    >
      <p
        className="c0"
      >
        color prop test
      </p>
    </Primitive>
  </Paragraph>
</TextPrimitive>
`;

exports[`<Text /> primitive should render correctly based on element prop 1`] = `
.c0 {
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  margin-bottom: 20px;
  text-decoration: none;
  font-size: 18px;
  line-height: 1.12;
}

<TextPrimitive
  element="h3"
>
  <Paragraph
    color="black"
    element="h3"
    size="regular"
    spacing="md"
    underline={false}
    weight="regular"
  >
    <Primitive
      className="c0"
      color="black"
      element="h3"
      size="regular"
      spacing="md"
      underline={false}
      weight="regular"
    >
      <h3
        className="c0"
      >
        element prop test
      </h3>
    </Primitive>
  </Paragraph>
</TextPrimitive>
`;

exports[`<Text /> primitive should render correctly based on size prop 1`] = `
.c0 {
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  margin-bottom: 20px;
  text-decoration: none;
  font-size: 14px;
  line-height: 1.12;
}

<TextPrimitive
  element="p"
  size="tiny"
>
  <Paragraph
    color="black"
    element="p"
    size="tiny"
    spacing="md"
    underline={false}
    weight="regular"
  >
    <Primitive
      className="c0"
      color="black"
      element="p"
      size="tiny"
      spacing="md"
      underline={false}
      weight="regular"
    >
      <p
        className="c0"
      >
        size prop test
      </p>
    </Primitive>
  </Paragraph>
</TextPrimitive>
`;

exports[`<Text /> primitive should render correctly based on underline prop 1`] = `
.c0 {
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  margin-bottom: 20px;
  text-decoration: underline;
  font-size: 18px;
  line-height: 1.12;
}

<TextPrimitive
  color="grey"
  element="p"
  underline={true}
>
  <Paragraph
    color="grey"
    element="p"
    size="regular"
    spacing="md"
    underline={true}
    weight="regular"
  >
    <Primitive
      className="c0"
      color="grey"
      element="p"
      size="regular"
      spacing="md"
      underline={true}
      weight="regular"
    >
      <p
        className="c0"
      >
        underline prop test
      </p>
    </Primitive>
  </Paragraph>
</TextPrimitive>
`;

exports[`<Text /> primitive should render correctly based on variant prop 1`] = `
.c0 {
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  margin-bottom: 20px;
  text-decoration: none;
  font-size: 26px;
  line-height: 28px;
}

@media (min-width:568px) {
  .c0 {
    font-size: 30px;
    line-height: 32px;
  }
}

@media (min-width:768px) {
  .c0 {
    font-size: 32px;
    line-height: 35px;
  }
}

<TextPrimitive
  element="h3"
  variant="h2"
>
  <H2
    color="black"
    element="h3"
    spacing="md"
    underline={false}
    weight="regular"
  >
    <Primitive
      className="c0"
      color="black"
      element="h3"
      spacing="md"
      underline={false}
      weight="regular"
    >
      <h3
        className="c0"
      >
        variant prop test
      </h3>
    </Primitive>
  </H2>
</TextPrimitive>
`;

exports[`<Text /> primitive should render correctly based on weight prop 1`] = `
.c0 {
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  margin-bottom: 20px;
  text-decoration: none;
  font-size: 18px;
  line-height: 1.12;
}

<TextPrimitive
  element="p"
  weight="light"
>
  <Paragraph
    color="black"
    element="p"
    size="regular"
    spacing="md"
    underline={false}
    weight="light"
  >
    <Primitive
      className="c0"
      color="black"
      element="p"
      size="regular"
      spacing="md"
      underline={false}
      weight="light"
    >
      <p
        className="c0"
      >
        weight prop test
      </p>
    </Primitive>
  </Paragraph>
</TextPrimitive>
`;

exports[`<Text /> primitive with default props 1`] = `
.c0 {
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  margin-bottom: 20px;
  text-decoration: none;
  font-size: 18px;
  line-height: 1.12;
}

<TextPrimitive>
  <Paragraph
    color="black"
    element="p"
    size="regular"
    spacing="md"
    underline={false}
    weight="regular"
  >
    <Primitive
      className="c0"
      color="black"
      element="p"
      size="regular"
      spacing="md"
      underline={false}
      weight="regular"
    >
      <p
        className="c0"
      >
        default prop test
      </p>
    </Primitive>
  </Paragraph>
</TextPrimitive>
`;

exports[`<TextPrimitive /> with initial props 1`] = `
.c0 {
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  margin-bottom: 20px;
  text-decoration: none;
  font-size: 18px;
  line-height: 1.12;
}

<TextPrimitive>
  <Paragraph
    color="black"
    element="p"
    size="regular"
    spacing="md"
    underline={false}
    weight="regular"
  >
    <Primitive
      className="c0"
      color="black"
      element="p"
      size="regular"
      spacing="md"
      underline={false}
      weight="regular"
    >
      <p
        className="c0"
      >
        Test initial
      </p>
    </Primitive>
  </Paragraph>
</TextPrimitive>
`;
